plugins/modules/system/homectl.py

```diff
@@ -0,0 +1,618 @@
+        self.gid = module.params['gid']
+        self.umask = module.params['umask']
+        self.memberof = module.params['memberof']
+        self.skeleton = module.params['skeleton']
+        self.shell = module.params['shell']
+        self.environment = module.params['environment']
+        self.timezone = module.params['timezone']
+        self.locked = module.params['locked']
+        self.passwordhint = module.params['passwordhint']
+        self.sshkeys = module.params['sshkeys']
+        self.language = module.params['language']
+        self.notbefore = module.params['notbefore']
+        self.notafter = module.params['notafter']
+        self.mountopts = module.params['mountopts']
+
+        self.result = {}
+
+    # Cannot run homectl commands if service is not active
+    def homed_service_active(self):
+        is_active = True
+        cmd = ["systemctl", "show", "systemd-homed.service", "-p", "ActiveState"]
+        rc, show_service_stdout, stderr = self.module.run_command(cmd)
+        if rc == 0:
+            state = show_service_stdout.rsplit("=")[1]
+            if state.strip() != "active":
+                is_active = False
+        return is_active
+
+    def user_exists(self):
+        # Get user properties if they exist in json
+        exists = False
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append("inspect")
+        cmd.append(self.name)
+        cmd.append("-j")
+        rc, stdout, stderr = self.module.run_command(cmd, use_unsafe_shell=True)
+        if rc == 0:
+            exists = True
+        return exists
+
+    def create_user(self):
+        record = self.create_json_record(create=True)
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append("create")
+        cmd.append("--identity=-")  # Read the user record from standard input.
+        return(self.module.run_command(cmd, data=record, use_unsafe_shell=True))
+
+    def _hash_password(self, password):
+        method = crypt.METHOD_SHA512
+        salt = crypt.mksalt(method, rounds=10000)
+        pw_hash = crypt.crypt(password, salt)
+        return pw_hash
+
+    def remove_user(self):
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append('remove')
+        cmd.append(self.name)
+        return(self.module.run_command(cmd, use_unsafe_shell=True))
+
+    def modify_user(self):
+        record = self.create_json_record()
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append('update')
+        cmd.append(self.name)
+        cmd.append('--identity=-')  # Read the user record from standard input.
+        # Resize disksize now resize = true
+        # This is not valid in user record (json) and requires it to be passed on command.
+        if self.disksize and self.resize:
+            cmd.append('--and-resize')
+            cmd.append('true')
+        return(self.module.run_command(cmd, data=record, use_unsafe_shell=True))
```

Comment:
I don't think this should be run except if a change is to be expected. (Also the resize operation does not seem to count for changed?)

Reply:
good catch fixing this.